driving_environment_analyzer/src/node.cpp

```diff
@@ -0,0 +1,394 @@
+      return true;
+    }
+  }
+
+  return false;
+}
+
+bool existCrosswalk(
+  const lanelet::ConstLanelets & lanes, const route_handler::RouteHandler & route_handler)
+{
+  const auto overall_graphs = route_handler.getOverallGraphPtr();
+  for (const auto & lane : lanes) {
+    constexpr int PEDESTRIAN_GRAPH_ID = 1;
+    if (!overall_graphs->conflictingInGraph(lane, PEDESTRIAN_GRAPH_ID).empty()) {
+      return true;
+    }
+  }
+
+  return false;
+}
+}  // namespace
+
+DrivingEnvironmentAnalyzer::DrivingEnvironmentAnalyzer(const rclcpp::NodeOptions & node_options)
+: Node("driving_environment_analyzer", node_options)
+{
+  using std::placeholders::_1;
+  using std::chrono_literals::operator""ms;
+
+  const auto bag_filename = declare_parameter<std::string>("bag_path");
+  const auto use_map_in_bag = declare_parameter<bool>("use_map_in_bag");
+
+  timer_ = rclcpp::create_timer(
+    this, get_clock(), 100ms, std::bind(&DrivingEnvironmentAnalyzer::analyze, this));
+
+  sub_map_ = create_subscription<HADMapBin>(
+    "input/lanelet2_map", rclcpp::QoS{1}.transient_local(),
+    std::bind(&DrivingEnvironmentAnalyzer::onMap, this, _1));
+
+  reader_.open(bag_filename);
+
+  if (!isEnoughData(use_map_in_bag)) {
+    rclcpp::shutdown();
+  }
+}
+
+bool DrivingEnvironmentAnalyzer::isEnoughData(const bool use_map_in_bag)
+{
+  const std::string topic_route = "/planning/mission_planning/route";
+  const std::string topic_map = "/map/vector_map";
+
+  rclcpp::Serialization<LaneletRoute> serializer_route;
+  rclcpp::Serialization<HADMapBin> serializer_map;
+
+  rosbag2_storage::StorageFilter filter;
+  filter.topics.emplace_back(topic_route);
+  filter.topics.emplace_back(topic_map);
+  reader_.set_filter(filter);
+
+  while (reader_.has_next()) {
+    const auto data = reader_.read_next();
+
+    if (data->topic_name == topic_route) {
+      rclcpp::SerializedMessage serialized_msg(*data->serialized_data);
+      const auto deserialized_message = std::make_shared<LaneletRoute>();
+      serializer_route.deserialize_message(&serialized_msg, deserialized_message.get());
+      route_msgs_.push_back(*deserialized_message);
+    }
+
+    if (data->topic_name == topic_map) {
+      rclcpp::SerializedMessage serialized_msg(*data->serialized_data);
+      const auto deserialized_message = std::make_shared<HADMapBin>();
+      serializer_route.deserialize_message(&serialized_msg, deserialized_message.get());
+      route_handler_.setMap(*deserialized_message);
+      has_map_data_ = true;
+    }
+  }
+
+  if (route_msgs_.empty()) {
+    RCLCPP_ERROR(get_logger(), "topic: %s is not included. shutdown.", topic_route.c_str());
+    return false;
+  }
+
+  if (use_map_in_bag && !has_map_data_) {
+    RCLCPP_ERROR(get_logger(), "topic: %s is not included. shutdown.", topic_map.c_str());
+    return false;
+  }
+
+  return true;
+}
+
+void DrivingEnvironmentAnalyzer::onMap(const HADMapBin::ConstSharedPtr msg)
+{
+  route_handler_.setMap(*msg);
+  has_map_data_ = true;
```

Comment:
nits:
route_handler_ and has_map_data_ can be converted to one variable by making route_handler_ std::optional?